### PR TITLE
fix: add ingress and refactor doc

### PR DESCRIPTION
**Pull request by aureamunoz**
For the moment this is the version manual, I will add tomorrow also a branch with the jib maven plugin and maybe other way that I have investigated today.

WDYT @cmoulliard?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,76 @@
 # hello-world-fwless
-Simple java API rest without framework
+Simple java rest API without framework.
 
+This branch aims to showcase how to use [Dekorate](https://github.com/dekorateio/dekorate) to deploy this microservice on a Kubernetes cluster.
```
Comment (cmoulliard) [inline README.md]: ... to generate the YAML resources and next deploy this microservice on a kubernetes cluster.

File: README.md [hunk 2]
```diff
@@ -1,12 +1,76 @@
...
 
-This branch aims to showcase how to use Dekorate to deploy this microservice on Openshift.
+## Set up
+This project uses [kind](https://kind.sigs.k8s.io/) for running a Kubernetes cluster locally. You can install it following the instructions [here](https://kind.sigs.k8s.io/docs/user/quick-start/#installation).
+Once installed, we need to set it up in order to use a local docker registry and install the ingress controller to make Ingress resources work. So follow the next steps:
```
Comment (cmoulliard) [inline README.md]: Once installed, we need to set it up in order to use a local docker registry and install the ingress controller to make Ingress resources work.

-->

Once installed, we need to set it up in order to :
- Use a local docker registry needed to provide the image of the microservice when the pod is created
- Install an Ingress controller managing the access from the host to the cluster of the services exposed

File: README.md [hunk 3]
```diff
@@ -1,12 +1,76 @@
...
+```
+
+2. Install the Ingress controller in the cluster.
+
+You can deploy the [NGINX Ingress Controller](https://github.com/kubernetes/ingress-nginx) running the following command:
+```
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/master/deploy/static/provider/kind/deploy.yaml
+```
+
+The environment is now ready to deploy our application using `Dekorate`.
```
Comment (cmoulliard) [inline README.md]: This is not correct. Dekorate only generates the MANIFEST/YAML resources but dont deploy them. Can you rephrase please ?

File: README.md [hunk 4]
```diff
@@ -16,45 +80,114 @@ Generating this manifests is very easy, you just need add the proper dependency
  </dependency>
 ```
 
-We also need to add an annotation to enable dekorate. In this case we will use `@KubernetesApplication` which also gives us access to more Kubernetes specific configuration options.
-Edit the App class and add the indicated annotation.
+We need to enable `Dekorate` by adding an annotation. In this case we will use `@KubernetesApplication` which also gives us access to more Kubernetes specific configuration options.
```
Comment (cmoulliard) [inline README.md]: To tell to Dekorate what should be populate during the creation of the MANIFEST for Kubernetes, we will add a Java annotation called `@KubernetesApplication` which ... 
Remark: Add here the link to the dekorate documentation please

File: README.md [hunk 5]
```diff
@@ -16,45 +80,114 @@ Generating this manifests is very easy, you just need add the proper dependency
...
+@KubernetesApplication(
+        name = "hello-world-fwless-k8s",        
+        ports = @Port(name = "web", containerPort = 8080),  
+        host = "fw-app.127.0.0.1.nip.io", //2
+        expose = true, //3
+        imagePullPolicy = ImagePullPolicy.Always //4
+)
 ```
-We need to specify a port in order to prevent dekorate it is about a Web Application, this way a service will be generated in the k8s manifest.
+- We need to specify a **`@Port`** in order to prevent dekorate it is about a Web Application, this way a service will be generated in the k8s manifest.
```
Comment (cmoulliard) [inline README.md]: This sentence is not correct in english and confusing. I suppose that you would like to say that we want to specify within the pod definition that the web application is running using the port : 8080 and this port will also be used to expose a service using the same value ...

File: README.md [hunk 6]
```diff
@@ -16,45 +80,114 @@ Generating this manifests is very easy, you just need add the proper dependency
...
 
-NOTE: if you are using minikube a trick that often works is to use a service of type NodePort.
```
Comment (cmoulliard) [inline README.md:27]: this trick is not specific to minikube and could be used with kind, minishift, ... ;-)

File: README.md [hunk 7]
```diff
@@ -16,45 +80,114 @@ Generating this manifests is very easy, you just need add the proper dependency
...
+- The host under which the application is going to be exposed.
+- **`expose`** controls whether the application should be exposed via Ingress.
```
Comment (cmoulliard) [inline README.md]: should be exposed via an Ingress route accessible from the host

File: README.md [hunk 8]
```diff
@@ -16,45 +80,114 @@ Generating this manifests is very easy, you just need add the proper dependency
...
+- We use **`Always`** in order to be able to use an updated image.
 
-Generate the manifests launching a project compilation. Navigate to the directory and run `mvn clean package`. The generated manifests can be found under `target/classes/META-INF/dekorate.
+Generate the manifests launching a project compilation. Navigate to the directory and run `mvn clean package`. The generated manifests can be found under `target/classes/META-INF/dekorate`.
```
Comment (cmoulliard) [inline README.md]: This is not correct too `Generate the manifests launching a project compilation` and must be rephrased

File: README.md [hunk 9]
```diff
@@ -16,45 +80,114 @@ Generating this manifests is very easy, you just need add the proper dependency
...
 
-NOTE: To perform the following steps you need to be connected to a running kubernetes cluster.
+## Building and deploying
 
-Run the following command to create the resources defined in the manifest.
+We need to somehow build and push the image to the registry to make it available for the kubernetes cluster.
```
Comment (cmoulliard) [inline README.md]: You should better introduce this part of the doc and mentions by example. `Now that we have populated YAML kubernetes resources able to deploy the application on the cluster, we must first create an container image and push it to a local container registry .... `

File: README.md [hunk 10]
```diff
@@ -16,45 +80,114 @@ Generating this manifests is very easy, you just need add the proper dependency
...
+docker tag USERNAME/hello-world-fwless:1.0-SNAPSHOT localhost:5000/hello-world-fwless:1.0-SNAPSHOT
 ```
-kubectl apply -f target/classes/META-INF/dekorate/kubernetes.yml
+Finally, you can push the image to the local registry
+```
+docker push localhost:5000/hello-world-fwless:1.0-SNAPSHOT
 ```
 
-At last, we need to expose the service in order to make it accessible from outside the cluster. For that, list the service using the command:
+As we have pushed the image to the local registry, we need to modify manually the kubernetes yml file generated to use that image.
```
Comment (cmoulliard) [inline README.md]: Add here a message in bold to warn the user as this is something that dekorate dont yet suppoprt and a manual modification is needed 

`**WARNING** : ....`

File: README.md [hunk 11]
```diff
@@ -16,45 +80,114 @@ Generating this manifests is very easy, you just need add the proper dependency
...
+image: USERNAME/hello-world-fwless:1.0-SNAPSHOT
+```
+by
 ```
-oc get services
+image: localhost:5000/hello-world-fwless:1.0-SNAPSHOT
 ```
 
-Copy the name of the service and expose it running:
+Finally, we will deploy the application by posting the manifest with the following command:
```
Comment (cmoulliard) [inline README.md]: ... we will deploy the application using the yaml resources generated under the namespace `demo` using the following command

File: README.md [hunk 12]
```diff
@@ -16,45 +80,114 @@ Generating this manifests is very easy, you just need add the proper dependency
...
 
 ```
-oc expose service hello-world-fwless
+kubectl create ns demo
+kubectl apply -f target/classes/META-INF/dekorate/kubernetes.yml -n demo
 ```
 
-Now, you should be able to get the address of the service and access the application with a browser. To get the adress, tape:
+Check the application by opening a browser to http://fw-app.127.0.0.1.nip.io/api/hello or curl http://fw-app.127.0.0.1.nip.io/api/hello
```
Comment (cmoulliard) [inline README.md]: After a few seconds, check if the application is running and available at the following address opened within your browser `` or do a curl/wget or httpie within a terminal

File: README.md [hunk 13]
```diff
@@ -16,45 +80,114 @@ Generating this manifests is very easy, you just need add the proper dependency
...
 
+Dekorate allows the user to trigger container image builds and deploy after the end of compilation, so the above actions can also be performed by executing the following command:
```
Comment (cmoulliard) [inline README.md]: As before, add a WARNING/IMPORTANT/REMARK... using `**WARNING/IMPORTANT/REMARK...**`, include the link to the dekorate doc explaining the dekorate build strategy ...

`**IMPORTANT**: Alternatively, you could also delegate the build of the container image using the following dekorate command ...`

File: README.md [hunk 14]
```diff
@@ -16,45 +80,114 @@ Generating this manifests is very easy, you just need add the proper dependency
...
 ```
-kubectl get services
+mvn clean package -Ddekorate.build=true  -Ddekorate.push=true -Ddekorate.docker.registry="localhost:5000" -Ddekorate.deploy=true
 ```
 
-Open a browser to x.x.x.x:y.y/api/hello
+### Using jib-maven-plugin
```
Comment (cmoulliard) [inline README.md:149]: Add a few words here to explain why JIB and advantages vs dockerfile build used before ;-)

File: README.md (hunk 3 above)
Comment (aureamunoz) [reply README.md]: ok

File: README.md (hunk 5 above)
Comment (aureamunoz) [reply README.md]: The Port tells to Dekorate that a service should be created. I will try to explain clearer

File: README.md [hunk 15]
```diff
@@ -84,7 +84,8 @@ Generating these manifests using `Dekorate` is very easy, you just need to add t
  </dependency>
 ```
 
-Next, we will add a Java annotation to tell to `Dekorate` what should be populate during the creation of the MANIFEST for Kubernetes. In this case we will use [`@KubernetesApplication`](https://github.com/dekorateio/dekorate#kubernetes) which also gives us access to more Kubernetes specific configuration options.
+Next, we will add a Java annotation for `Dekorate` to tune the generated manifest. It's possible [to configure Dekorate using](https://github.com/dekorateio/dekorate#usage) Java annotations, configuration properties (application.properties), both.
```
Comment (cmoulliard) [inline README.md]: Is this concept well described here `generated manifest` vs `yaml resources` ?
